port onchanges to API 8.0 to fix #22

yvaucher opened the pull request.

yvaucher committed:
port onchanges to API 8.0 to fix #22

yvaucher commented:
I have to refactor this a bit -> WIP

yvaucher commented on framework_agreement_sourcing/model/logistic_requisition_source.py:
```diff
@@ -289,50 +288,64 @@ def onchange_sourcing_method(self, cr, uid, ids, method, req_line_id,
+                'supplier_id': agreement.supplier_id.id}
+        self.write(vals)
+        res = {}
         if not enough_qty:
```
This check seems useless as in framework_agreement get_cheapest_agreement_for_qty only returns an agreement if it found a matching one with enough qty.

@nbessi @lepistone  have we somewhere in OCA modules an override of get_cheapest_agreement_for_qty that could return something else than:

(None, None) or (agreement_id, True) ?

lepistone replied:
@yvaucher I don't know off the top of my head. In any case, ping me if I can help.

yvaucher committed:
Add a second search when no agreement is found

This restore former warning message when we can find an agreement with
not enough quantities
This allow to find and finish the framework agreement almost finished.
As it is preferable to finish agreements completely instead of
contracting new ones and be left with multiple remainings of agreements.

foutoucour commented:
Do you plan to use openerp.fields classes to have a full port to v8.0?
It seems the new api is getting better and better.

also you are using osv and orm. They are a redondant, aren't they?

yvaucher commented:
@foutoucour I think porting fields in this PR is out of scope of the fix. And odoo 8.0 old api is not yet deprecated. :smiley: 

This PR does not intend to port the whole code to v8.0. I just fixed the onchanges tooking the opportunity to port them to new API.
I agree that v8.0 api is getting better. Nevertheless, after several issues with it, we decided not to port to v8.0 when not necessary.

About `osv` and `orm`:

They are not redundant, as I use direct path for old style fields:
https://github.com/odoo/odoo/blob/8.0/openerp/osv/fields.py

This to use new fields in onchanges. (just seeing I'm calling the wrong datetime BTW)

And didn't changed the old stylish path to Model (which is in fact in models.py now but this is out of scope of this PR)
https://github.com/odoo/odoo/blob/8.0/openerp/osv/orm.py

lepistone commented:
Thanks! :+1:

foutoucour commented:
:+1: 
Thanks for the port!
